instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/AsyncInstrumentRegistry.java

```diff
@@ -21,14 +19,28 @@
 import javax.annotation.Nullable;
 
 // TODO: refactor this class, there's too much copy-paste here
-final class AsyncInstrumentRegistry {
+public final class AsyncInstrumentRegistry {
```

Comment:
can you add a comment on this class about why it needs to live in the bootstrap class loader?